Shell/Shell.h

```diff
@@ -247,7 +254,7 @@ class Shell : public Core::Object {
     };
 
     HashMap<String, ShellFunction> m_functions;
-    Vector<LocalFrame> m_local_frames;
+    Vector<OwnPtr<LocalFrame>> m_local_frames;
```

Comment:
If you don't intend to store null values, use `NonnullOwnPtr` instead of `OwnPtr`.
You can also use `NonnullOwnPtrVector` for extra serenityness :)

Reply:
Ah yes, that one pattern I always forget.